flutter/lib/models/input_model.dart

```diff
@@ -1213,13 +1213,18 @@ class InputModel {
     }
 
     if (!cursorModel.gotMouseControl) {
-      bool selfGetControl =
-          (x - lastMousePos.dx).abs() > kMouseControlDistance ||
-              (y - lastMousePos.dy).abs() > kMouseControlDistance;
+      final dx = x - lastMousePos.dx;
+      final dy = y - lastMousePos.dy;
+
+      // Use the distance squared to avoid sqrt()
+      final d2 = dx * dx + dy + dy;
+      final threshold2 = kMouseControlDistance * kMouseControlDistance;
+
+      final bool selfGetControl = d2 > threshold2;
+
       if (selfGetControl) {
         cursorModel.gotMouseControl = true;
       } else {
-        lastMousePos = ui.Offset(x, y);
```

Comment:
This is intented.


>  local control is only regained if `kMouseControlDistance` is exceeded in a single mouse event.

As you guessed,
the user on the controlling side has to move the mouse a significant amount (in a short time) to take control.